var/spack/repos/builtin/packages/hpx/package.py

```diff
@@ -83,16 +83,23 @@ class Hpx(CMakePackage, CudaPackage):
     # https://github.com/STEllAR-GROUP/hpx/issues/4728#issuecomment-640685308
     depends_on('boost@:1.72.0', when='@:1.4')
 
+    depends_on('boost@1.64.0:', when='@1.5')
+    depends_on('boost@1.66.0:', when='@1.6')
+
     depends_on('boost+filesystem', when='cxxstd=11')
     depends_on('boost+filesystem', when='cxxstd=14')
-    depends_on('boost+system+program_options+chrono+date_time+log+regex+thread')
+    depends_on('boost+filesystem', when='^gcc@:8')
+    depends_on('boost+filesystem', when='^clang@:8')
```

Comment:
Is this correct syntax for requiring `filesystem` only for certain compiler versions? I think the check for clang should in theory even be on the stdlib, since one can use clang >= 9 with libstdc++ and headers that don't have `std::filesystem`, but I've no clue how to express that in spack.

Reply:
```suggestion
    depends_on('boost+filesystem', when='%gcc@:8')
    depends_on('boost+filesystem', when='%clang@:8')
```
You might also want to add `apple-clang`, but I'm not sure about their version numbers

Reply:
But yeah, this is probably the best we can do right now.

Reply:
Thanks for the fixes to the compiler spec, I always get those wrong.